src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateNoticeOfProceeding.java

```diff
@@ -58,8 +63,10 @@
                 NOTICE_OF_PROCEEDINGS_DOCUMENT_NAME
             );
 
-        } else if (!isSoleApplication && (!isApplicant1Represented || !isApplicant2Represented)) {
-            if (!isApplicant1Represented) {
+        } else if (!isSoleApplication
```

Comment:
I think you could simplifiy this to:

if (isSoleApplication && && !isApplicant1Represented) { ... stuff }
if (!isSoleApplication && (!isApplicant1Represented || isApplicant1Offline) { ... NoP App1 }
if (!isSoleApplication && (!isApplicant2Represented || isApplicant2Offline) { ... NoP App2 }

then it would be a bit less nested